internal/webhook/multiclusterservice_webhook_test.go

```diff
@@ -0,0 +1,264 @@
+// limitations under the License.
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/Mirantis/hmc/api/v1alpha1"
+	"github.com/Mirantis/hmc/internal/utils"
+	"github.com/Mirantis/hmc/test/objects/multiclusterservice"
+	"github.com/Mirantis/hmc/test/objects/template"
+	"github.com/Mirantis/hmc/test/scheme"
+)
+
+const (
+	testMCSName          = "testmcs"
+	testSvcTemplate1Name = "test-servicetemplate-1"
+	testSvcTemplate2Name = "test-servicetemplate-2"
+)
+
+func TestMultiClusterServiceValidateCreate(t *testing.T) {
+	ctx := admission.NewContextWithRequest(context.Background(), admission.Request{
+		AdmissionRequest: admissionv1.AdmissionRequest{
+			Operation: admissionv1.Create,
+		},
+	})
+
+	tests := []struct {
+		name            string
+		mcs             *v1alpha1.MultiClusterService
+		existingObjects []runtime.Object
+		err             string
+		warnings        admission.Warnings
+	}{
+		{
+			name: "should fail if the ServiceTemplates are not found in hmc-system namespace",
+			mcs: multiclusterservice.NewMultiClusterService(
+				multiclusterservice.WithName(testMCSName),
+				multiclusterservice.WithServiceTemplate(testSvcTemplate1Name),
+			),
+			existingObjects: []runtime.Object{
+				template.NewServiceTemplate(
+					template.WithName(testSvcTemplate1Name),
+					template.WithNamespace("othernamespace"),
+					template.WithValidationStatus(v1alpha1.TemplateValidationStatus{Valid: true}),
+				),
+			},
+			err: fmt.Sprintf("the MultiClusterService is invalid: servicetemplates.hmc.mirantis.com \"%s\" not found", testSvcTemplate1Name),
+		},
+		{
+			name: "should fail if the ServiceTemplates were found but are invalid",
+			mcs: multiclusterservice.NewMultiClusterService(
+				multiclusterservice.WithName(testMCSName),
+				multiclusterservice.WithServiceTemplate(testSvcTemplate1Name),
+			),
+			existingObjects: []runtime.Object{
+				template.NewServiceTemplate(
+					template.WithName(testSvcTemplate1Name),
+					template.WithNamespace(utils.DefaultSystemNamespace),
+					template.WithValidationStatus(v1alpha1.TemplateValidationStatus{
+						Valid:           false,
+						ValidationError: "validation error example",
+					}),
+				),
+			},
+			err: "the MultiClusterService is invalid: the template is not valid: validation error example",
+		},
+		{
+			name: "should succeed",
+			mcs: multiclusterservice.NewMultiClusterService(
+				multiclusterservice.WithName(testMCSName),
+				multiclusterservice.WithServiceTemplate(testSvcTemplate1Name),
+			),
+			existingObjects: []runtime.Object{
+				template.NewServiceTemplate(
+					template.WithName(testSvcTemplate1Name),
+					template.WithNamespace(utils.DefaultSystemNamespace),
```

Comment:
Let's overwrite the default system namespace in tests to check that the validation works properly if HMC is installed in another namespace.